src/Components/Sidebar/CategoryFilter/CategoryFilter.jsx

```diff
@@ -1,29 +1,50 @@
 import axios from "axios";
 import { useState, useEffect } from "react";
+import { useFilterReducerContext } from "../../../context/FilterReducer.context";
+import { Loader } from "../../Loader/Loader";
 
 export const CartegoryFilter = () => {
+  const { loading, categories, dispatch } = useFilterReducerContext();
+
   const [categoryData, setCategoryData] = useState([]);
 
+  // initializing categories from data
   useEffect(() => {
     (async function () {
       const { data } = await axios.get("/api/categories");
       setCategoryData(data.categories);
     })();
   }, []);
-  
+
   return (
     <>
       <li className="filter-section-title">Categories</li>
-      {categoryData.map((category) => {
-        return (
-          <li key={category.id}>
-            <label className="form-label">
-              <input type="checkbox" />
-              {category.categoryName}
-            </label>
-          </li>
-        );
-      })}
+      {loading ? (
+        <Loader />
+      ) : (
+        <>
```

Comment:
Also, there is no need to use fragment here. As on line 22, we have started a JavaScript code block, using `{` so we can directly use map.

Like, instead of
```javascript
) : (<>{}</>)
```

Do,
```javascript
) : (Array.map())
```

Reply:
Thanks for pointing this out !